### PR TITLE
feat: Add hint about renaming of circles to teams

### PR DESCRIPTION
**Pull request by juliushaertl**
Add hint about renaming of circles to teams to the developer upgrade guide for 29

### CODE REVIEW

**Review by ChristophWurst — COMMENTED** [review]
Should go into general because there could also be strings in the backend that need adjustment

**Comment by juliushaertl** [comment]
We would not rename APIs and internal references for now, that's why I moved it to frontend, but of course if backend provides string to the user then those also need adjustments, but I can move it.

File: developer_manual/app_publishing_maintenance/app_upgrade_guide/upgrade_to_29.rst [hunk 1]
```diff
@@ -19,6 +19,8 @@ Make sure your ``appinfo/info.xml`` allows for Nextcloud 29.
 Front-end changes
 -----------------
 
+* The Circles app will be renamed to Teams, meaning that any usage of the terms Circle/Circles need to get adjusted to Team/Teams for apps that make use of it, e.g. when sharing to a circle
```
Comment (max-nextcloud) [inline developer_manual/app_publishing_maintenance/app_upgrade_guide/upgrade_to_29.rst:22]: Trying to break the long sentence into multiple: 
```suggestion
* The Circles app will be renamed to Teams. Apps that use the terms Circle/Circles need to get adjusted to use Team/Teams instead. For example ``share to circle`` would become ``share to team``.
```

**Review by max-nextcloud — APPROVED** [review]
Thanks for taking care of this. Content is fine. I'll propose a different wording but good to go from my side either way.

**Comment by ChristophWurst** [comment]
> We would not rename APIs and internal references for now, that's why I moved it to frontend, but of course if backend provides string to the user then those also need adjustments, but I can move it.

I mean things like l10n strings in activities, notifications, php templates for settings

**Comment by juliushaertl** [comment]
Moved and applied the wording suggestion.

**Pull request merged**